connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/connect/client/SparkConnectClient.scala

```diff
@@ -58,10 +58,13 @@ private[sql] class SparkConnectClient(
    * @return
    *   A [[proto.AnalyzePlanResponse]] from the Spark Connect server.
    */
-  def analyze(request: proto.AnalyzePlanRequest): proto.AnalyzePlanResponse =
+  def analyze(request: proto.AnalyzePlanRequest): proto.AnalyzePlanResponse = {
+    artifactManager.uploadAllClassFileArtifacts()
```

Comment:
Does this method cache previous scanned artifacts? If not maybe the overhead of calling analyze and execute becomes a bit large?

Reply:
> Does this method cache previous scanned artifacts?

Good eye but not at the moment, no. However, the plan is to implement this later as an optimisation.